### DISCUSSION TITLE
How to override part of lualine config

### DISCUSSION THREAD

**Discussion by grifferz** [LazyVim/LazyVim#3709] [Q&A] [answered]
Please excuse my lack of Lua knowledge.

On the lualine I'd like to have the textual filetype as well as the icon. I see the default config for lualine has:

```lua
lualine_c = {
  LazyVim.lualine.root_dir(),
  {
    "diagnostics",
    symbols = {
      error = icons.diagnostics.Error,
      warn = icons.diagnostics.Warn,
      info = icons.diagnostics.Info,
      hint = icons.diagnostics.Hint,
    },
  },
  { "filetype", icon_only = true, separator = "", padding = { left = 1, right = 0 } },
  { LazyVim.lualine.pretty_path() },
},
```

So I think I just need to override that `icon_only = true` to be `icon_only = false`. But how to do that without copying the entire config from https://www.lazyvim.org/plugins/ui#lualinenvim into e.g `~/.config/nvim/lua/plugins/lualine.lua`?

I get I can do something like this:

```lua
return {
  "nvim-lualine/lualine.nvim",

  opts = function(_ opts)
    for _, component in ipairs(opts.sections.lualine_c) do
      if component[1] == "filetype" then
        -- do something with component here
      end
    end
  end,
}
```

but I don't know what the "do something" bit would be here. Or even if this is the right way to go with this.

Any hints would be appreciated!

**Comment by rubiin** [comment]
You can use the table method I guess since the sections are basically a table

https://www.educative.io/answers/what-is-the-tableinsert-method-in-lua

**Reply by rubiin** [reply]
```lua

local  lualine_c = {'branch', 'diff', 'diagnostics'}


table.insert(lualine_c,1, "someval")

```

**Comment by dpetka2001** [comment, marked as answer]
```lua
for _, component in ipairs(opts.sections.lualine_c) do
  if component[1] == "filetype" then
	component.icon_only = false
  end
end
```

**Reply by grifferz** [reply]
Ah, that simple! Thank you!
